src/open_inwoner/cms/open_product/product_list/cms_plugins.py

```diff
@@ -0,0 +1,122 @@
+from datetime import date
+
+from django.core.cache import cache as django_cache
+from django.utils.translation import gettext_lazy as _
+
+from cms.plugin_base import CMSPluginBase
+from cms.plugin_pool import plugin_pool
+
+from open_inwoner.open_product.models import OpenProductConfig
+
+from .forms import ProductListForm
+from .models import ProductList
+from .utils import get_open_product_client
+
+
+@plugin_pool.register_plugin
+class ProductListPlugin(CMSPluginBase):
+    module = _("Open Product")
+    name = _("List of products")
+    model = ProductList
+    form = ProductListForm
+    render_template = "cms/product_list/product_list_plugin.html"
+    cache = False
+
+    # The amount of days before an indicator is shown
+    #  that the product/document is about to expire.
+    EXPIRATION_WARNING_TIME_SLOT = 14
+
+    def __init__(self, model=None, admin_site=None):
+        super().__init__(model, admin_site)
+        self._client = None
+
+        # 1 hour intervals for caching producttypes data
+        self.PRODUCTTYPES_CACHE_TIMEOUT = 3600
+
+    # Needs to be a property for the tests to work
+    @property
+    def client(self):
+        if self._client is None:
+            self._client = get_open_product_client()
+        return self._client
+
+    def _get_cached_producttypes(self, theme):
+        """
+        Retrieves cached producttypes. If not available,
+         they are retrieved from the API.
+        """
+
+        cache_id = f"products:{theme}"
+        cached_producttypes = django_cache.get(cache_id)
+
+        if not cached_producttypes:
+            producttypes = self.client.list_product_types(themas__uuid=theme)["results"]
+
+            # Retrieve local configuration of product actions
+            producttypeactionsconfig = OpenProductConfig.get_solo()
+            action_urls = producttypeactionsconfig.action_urls
+
+            # For every producttype, insert local 'action_url' configuration
+            for pt in producttypes:
+                for key, value in action_urls.items():
+                    action_full_name = key.split(":")
+                    pt_name = action_full_name[0]
+                    action_name = action_full_name[1]
+
+                    if pt["naam"] == pt_name:
+                        for action in pt["acties"]:
+                            if action["naam"] == action_name:
+                                action["action_url"] = value
+
+            django_cache.set(cache_id, producttypes, self.PRODUCTTYPES_CACHE_TIMEOUT)
+            return producttypes
+        else:
+            return cached_producttypes
```

Comment:
```suggestion
        if cached_producttypes:
            return cached_producttypes
            
        producttypes = self.client.list_product_types(themas__uuid=theme)["results"]
        ...
        return producttypes
```
Reduce nesting by reverting the order; makes code easier to read.